test/NuGet.Core.Tests/NuGet.Common.Test/PathResolverTests.cs

```diff
@@ -0,0 +1,703 @@
+
+            Verify(new[]
+                {
+                    @"\dir1\dir2\file1.txt",
+                    @"\dir1\dir2\file2.txt",
+                    @"\dir1\dir4\file1.txt",
+                    @"\dir1\dir4\file2.txt"
+                }, actualFullPaths);
+        }
+
+        [Platform(Platform.Darwin)]
+        [Theory]
+        [InlineData("dir1/dir*/*.txt", new[]
+            {
+                "/dir1/dir2/file1.txt",
+                "/dir1/dir2/file2.txt",
+                "/dir1/dir4/file1.txt",
+                "/dir1/dir4/file2.txt"
+            })]
+        [InlineData(@"dir1\dir*\*.txt", new string[] { })]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryPatternAndFileNamePatternRecursivelyFindsAllMatchingFiles_OnMacOs(string searchPath, string[] expectedResults)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(expectedResults, actualFullPaths);
+        }
+
+        [Platform(Platform.Linux)]
+        [Theory]
+        [InlineData("dir1/dir*/*.txt", new[]
+            {
+                "/dir1/dir2/file1.txt",
+                "/dir1/dir2/file2.txt",
+                "/dir1/dir4/file1.txt",
+                "/dir1/dir4/file2.txt"
+            })]
+        [InlineData(@"dir1\dir*\*.txt", new string[] { })]
+        public void PathResolver_PerformWildcardSearch_WithDirectoryPatternAndFileNamePatternRecursivelyFindsAllMatchingFiles_OnLinux(string searchPath, string[] expectedResults)
+        {
+            var actualFullPaths = PathResolver.PerformWildcardSearch(_fixture.TestDirectoryPath, searchPath);
+
+            Verify(expectedResults, actualFullPaths);
+        }
+
+        private void Verify(IEnumerable<string> expectedRelativePaths, IEnumerable<string> actualFullPaths)
+        {
+            var actualRelativePaths = actualFullPaths.Select(fullPath => fullPath.Substring(_fixture.TestDirectoryPath.Length));
+            var expectedResults = expectedRelativePaths.OrderBy(path => path);
+            var actualResults = actualRelativePaths.OrderBy(path => path);
+
+            Assert.Equal(expectedResults, actualResults);
+        }
+
+        private static IEnumerable<string> GetPlatformSpecificPaths(IEnumerable<string> platformUnspecificPaths)
+        {
+            return platformUnspecificPaths.Select(path => string.Format(path, Path.DirectorySeparatorChar));
+        }
+    }
+
+    /*
+    Test directory contents:
+
+        dir1
+            dir2
+                dir3
+                    file1.txt
+                    file2.txt
+                file1.txt
+                file2.txt
+            dir4
+                file1.txt
+                file2.txt
+            file1.txt
+            file2.txt
+        dir5
+        file1.txt
+        file2.txt
+    */
+    public sealed class TestDirectoryFixture : IDisposable
+    {
+        private DirectoryInfo _directory;
```

Comment:
Same as below, can this share some implementation with either the existing `TestDirectory` or the other `TestDirectory` below?